Fix EOM description in certificate validator helpers docs (#47)
squid-cache/squid-cache.github.io: docs/Features/AddonHelpers.md

This interface is similar to the SSL certificate generation interface.

> Request messages sent by Squid have the following syntax:

Reviewer: `"Input line received from Squid"` is text shared by all helper API documentation. Please revert.

Author: Reverting this change would make things worse overall. I will not do it. We should describe certificate helper messages without using line-based terminology that is not applicable to (and dangerously misleading for) these helpers. If we want to use the same terms for all helpers, then line-based helpers should be updated as well (and I can volunteer to do that), but that update is out of this PR scope.

>     [request-ID SP] action SP size SP body LF

Reviewer: Multiple issues:

1) Field representing "concurrency channel" is already defined for other helpers as `channel-ID`.
  a) Please use that name for consistency.
  b) AFAIK, concurrency channel is not an optional field for this helper.

2) This part of the documentation is a fieldset semantic definition, **not** binary syntax definition. As such the delimiter (`SP`) and EOL (`LF`) characters should not be listed. (Updating the entire page to define the binary grammar is topic for a different PR scope).

3) `body` is a vague term (used only by Factory, and only for this helper API). The existing text already provides a more-specific term `kv-pairs` which clearly documents that Squid sends a set of "key=value" fields. This term and usage is shared with all other helper APIs.

Please match the definition from PR #30:
```suggestion
    channel-ID action size [kv-pairs]
```

Author: > AFAIK, concurrency channel is not an optional field for this helper.

Why do you think so?

> 2. This part of the documentation is a fieldset semantic definition, **not** binary syntax definition. As such the delimiter (`SP`) and EOL (`LF`) characters should not be listed. (Updating the entire page to define the binary grammar is topic for a different PR scope).

I disagree that we should use what you call "fieldset semantic definition" for all helpers. I disagree that we should not list delimiters when describing certificate helper messages syntax, especially when other helpers are using the same delimiters differently. We should address certificate helper needs using the best tools matching those needs instead. IMO, this PR is a step forward in that direction.

... where `SP` is an ASCII space character, `LF` is an ASCII new line
character (\\n), and the other fields are documented below.

:warning: The `body` field usually contains new line characters. Use the
`size` field to find the end of the body.

- request-ID: An unsigned positive 64-bit decimal number uniquely identifying
  this request among all requests submitted to this helper process (i.e. on

>   this helper stdin connection). Squid sends this field when (and only when)
>   the `concurrency=N` helper configuration option is used with `N` values
>   grater than 1. When this field is present, the corresponding helper response
>   must start with a matching `request-ID` field.

Reviewer: IIRC these certificate helpers were where we chose to make concurrency the default-on and a mandatory field. Which would mean all this text about special values is wrong/irrelevant, at least for this specific helper API.

Please use the text already agreed in PR #30:
```suggestion
- channel-ID
:  A numeric identifier for the concurrency channel correlating this message and its response.
   Helper should typically treat this as an opaque value that must be returned as the start of the response to Squid.
   Multi-threaded helpers must ensure that each message is written back to Squid as an "atomic" sequence of bytes,
   this parameter permits responses to be returned out-of-order to squid.
```

Author: > IIRC these certificate helpers were where we chose to make concurrency the default-on and a mandatory field.

Where? `sslcrtvalidator_children` documents `concurrency=...` option just like other helpers that support that optional feature. Moreover, that helper defaults to _no_ concurrency:

```
NAME: sslcrtvalidator_children
DEFAULT: 32 startup=5 idle=1 concurrency=1
```

> - action: A string with the name of the helper operation being requested.
>   Squid currently only sends `cert_validate` requests.

Reviewer: If you are going to change the field semantic from "request" to "action". Then please do so consistently.

```suggestion
- action
:   The type of action being requested. Presently the code
    `cert_validate` is the only action sent.
```

... where `SP` is an ASCII space character, `SOH` is an ASCII "Start of
Heading" character (with the value of 1), and the other fields are documented
below.

:warning: If the response `body` field is empty or contains a single key=value
pair that has no embedded new lines, then the entire response message should
have no new line characters.

- request-ID: An unsigned positive 64-bit decimal number equal to the
  `request-ID` field of the corresponding helper request. These fields must be
  sent when (and only when) the `concurrency=N` helper configuration option is
  used with `N` values grater than 1.

- result: A string matching one of the following codes:
        
        | code| code meaning                               |
        | --- | ------------------------------------------ |
        | OK  | Success. Certificate validated.            |
        | ERR | Success. Certificate not validated.        |
        | BH  | Failure. The helper encountered a problem. |
        
- size: A decimal number representing the size of the `body` field in bytes.
  The size of an empty `body` is zero.

- body: A possibly empty list of the following key=value pairs separated by

>   ASCII new line characters (\\n).

Reviewer: Please do not add the new term "body". It is far too easily conflated with the entire message being a "body". This area of the message is an optional sequence of `kv-pair`, as already documented.

        | key                     | value description           |
        | ----------------------- | --------------------------- |
        | cert\_***ID***          | A certificate send from helper to squid. The **ID** is an index number for this certificate                               |
        | error\_name\_***ID***   | The openSSL error name for the error **ID**                                                                               |
        | error\_reason\_***ID*** | A reason for the error **ID**                                                                                             |
        | error\_cert\_***ID***   | The broken certificate. It can be one of the certificates sent by helper to squid or one of those sent by squid to helper |
        

> Example response message (with the terminating SOH character shown as `^A`):

Reviewer: Abbreviations for ASCII character are standardized. SOH and EOL are specific characters with specific well-known semantic meanings.
 * The semantic meaning Squid is sending does not match `SOH` definition, and
 * The character(s) Squid is sending does not match the intended EOL definition.

The correct "term of art" to be using here is `EOM` ("END of MESSAGE").

```suggestion
Example response message (with the terminating EOM character sequence shown as `^A`):
```
